src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -0,0 +1,202 @@
+package com.cloudbees.jenkins.plugins.sshagent;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHUserPrivateKey;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.google.inject.Inject;
+import hudson.EnvVars;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.*;
+
+import javax.annotation.CheckReturnValue;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.*;
+
+public class SSHAgentStepExecution extends AbstractStepExecutionImpl {
+
+    private static final long serialVersionUID = 1L;
+
+    @StepContextParameter
+    private transient TaskListener listener;
+
+    @StepContextParameter
+    private transient Run build;
+
+    @StepContextParameter
+    private transient Launcher launcher;
+
+    @Inject
```

Comment:
(Since you seem to use it from `initRemoteAgent`, which is called from `onResume`, probably it should _not_ be `transient`. It must still be `optional` according to Guice; see newly edited `AbstractStepExecutionImpl` Javadoc.)